specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/examples/amlFilesystems_CreateOrUpdate.json

```diff
@@ -0,0 +1,144 @@
+            "sourceVault": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/some-rg/providers/Microsoft.KeyVault/vaults/examplekv"
+          }
+        },
+        "maintenanceWindow" : "2020-09-13T13:15Z",
+        "HSMSettings" : {
+          "container" : "https://somecontainer.blob.core.windows.net/containername",
+          "containerKeyVaultKey" : {
+            "keyUrl" : "https://somekv.vault.azure.net/keys/kvk/3540a47df75541378d3518c6a4bdf5af",
+            "sourceVault" : "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/some-rg/providers/Microsoft.KeyVault/vaults/somekv"
+          },
+          "importPrefix" : "asc",
+          "exportPrefix" : "asc"
+        }
+      },
+      "sku": {
+        "name": "Standard_2G"
+      }
+    }
+  },
+  "responses": {
+    "201": {
+      "body": {
+        "tags": {
+          "Dept": "ContosoAds"
+        },
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scgroup/providers/Microsoft.StorageCache/amlFilesystems/fs1",
+        "location": "westus",
+        "name": "fs1",
+        "type": "Microsoft.StorageCache/amlFilesystem",
+        "properties": {
+          "storageCapacityTiB": 3072,
+          "availabilityZones" : ["1", "2"],
+          "health": {
+            "state": "Healthy",
+            "statusDescription": "amlFilesystem is ok."
+          },
+          "provisioningState": "Succeeded",
+          "subnet": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scgroup/providers/Microsoft.Network/virtualNetworks/scvnet/subnets/sub1",
+          "encryptionSettings": {
+            "keyEncryptionKey": {
+              "keyUrl": "https://examplekv.vault.azure.net/keys/kvk/3540a47df75541378d3518c6a4bdf5af",
+              "sourceVault": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/some-rg/providers/Microsoft.KeyVault/vaults/examplekv"
+            }
+          },
+          "maintenanceWindow" : "2020-09-13T13:15Z",
+          "maintenanceStatus": {
+            "currentFirmwareVersion": "5.3.23",
+            "firmwareUpdateStatus": "available",
+            "firmwareUpdateDeadline": "2019-04-21T18:25:43.511Z",
+            "lastFirmwareUpdate": "2019-01-21T18:25:43.511Z",
+            "pendingFirmwareVersion": "5.3.24"
+          },
+          "hsmSettings" : {
+            "container" : "https://somecontainer.blob.core.windows.net/containername",
+            "containerKeyVaultKey" : {
+              "keyUrl" : "https://somekv.vault.azure.net/keys/kvk/3540a47df75541378d3518c6a4bdf5af",
+              "sourceVault" : "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/some-rg/providers/Microsoft.KeyVault/vaults/somekv"
+            },
+            "importPrefix" : "aml",
+            "exportPrefix" : "aml"
+          },
+          "namespaceHydrationStatus": {
+            "state": "NotStarted",
```

Comment:
Where do I find the list of possible values for this?

Reply:
in the definition in storagecache.json.
"amlFilesystemHydrationStatus" : {
      "description" : "Information about the amlFilesystem hydration status",
      "type" : "object",
      "readOnly" : true,
      "properties": {
        "state" : {
          "description" : "The state of the namespace hydration",
          "type" : "string",
          "enum" : [
            "NotStarted",
            "InProgress",
            "Failed",
            "Complete"
          ],
          "x-ms-enum": {
            "name": "HydrationStatusType",
            "modelAsString": true
          }
        },